Feature/controllers migration to kotlin

ajax-huzei-a opened the pull request.

ajax-huzei-a committed:
00001: KOTLIN MIGRATION
CandidateController migrate to Kotlin

ajax-huzei-a committed:
Rename .java to .kt

ajax-huzei-a committed:
00001: KOTLIN MIGRATION
CoordinatorController migrate to Kotlin and refactor some method names and CandidateController

ajax-huzei-a committed:
Rename .java to .kt

ajax-huzei-a committed:
Rename .java to .kt

ajax-huzei-a committed:
00001: KOTLIN MIGRATION
GuestController migrate to Kotlin

ajax-huzei-a committed:
Rename .java to .kt

ajax-huzei-a committed:
00001: KOTLIN MIGRATION
InterviewerController migrate to Kotlin and refactor some java classes

ajax-huzei-a committed:
00001: KOTLIN MIGRATION
JwtAuthenticationController migrate to Kotlin and refactor some classes

ajax-huzei-a committed:
Rename .java to .kt

ajax-huzei-a committed:
00001: KOTLIN MIGRATION
Add logger into JwtAuthenticationController

ajax-gnatiuk-s commented on src/main/java/intellistart/interviewplanning/controllers/CandidateController.kt:
```diff
@@ -0,0 +1,112 @@
+        authentication: Authentication
+    ): CandidateSlot {
+        val jwtUserDetails = authentication.principal as JwtUserDetails
+        val candidateSlot = CandidateSlot()
+        candidateSlot.date = candidateSlotDto.date
+        candidateSlot.period = periodService
+            .obtainPeriod(candidateSlotDto.from, candidateSlotDto.to)
+        candidateSlot.email = jwtUserDetails.email
+        candidateSlot.name = jwtUserDetails.name
+        return candidateSlot
```
```suggestion
        return CandidateSlot().apply {
            date = candidateSlotDto.date
            period = periodService.obtainPeriod(candidateSlotDto.from, candidateSlotDto.to)
            email = jwtUserDetails.email
            name = jwtUserDetails.name
        }
```

ajax-gnatiuk-s commented on src/main/java/intellistart/interviewplanning/controllers/CoordinatorController.kt:
```diff
@@ -0,0 +1,221 @@
+        return ResponseEntity.ok(savedBooking.toDto())
+    }
+
+
```
```suggestion
```

ajax-gnatiuk-s commented on src/main/java/intellistart/interviewplanning/controllers/CoordinatorController.kt:
```diff
@@ -0,0 +1,221 @@
+    }
+
+    private fun getFromDto(bookingDto: BookingDto): Booking {
+        val booking = Booking()
+        booking.subject = bookingDto.subject
+        booking.description = bookingDto.description
+
+        booking.interviewerSlot = interviewerSlotService.getById(bookingDto.interviewerSlotId)
+        booking.candidateSlot = candidateSlotService.getById(bookingDto.candidateSlotId)
+
+        booking.period = periodService.obtainPeriod(
+            bookingDto.from,
+            bookingDto.to
+        )
+
+        return booking
```
could be beautified with `.apply` function :)

ajax-gnatiuk-s commented on src/main/java/intellistart/interviewplanning/controllers/GuestController.kt:
```diff
@@ -0,0 +1,21 @@
+ */
+@RestController
+@CrossOrigin
+class GuestController(private val weekService: WeekService) {
+    @GetMapping("weeks/current")
```
```suggestion
class GuestController(private val weekService: WeekService) {

    @GetMapping("weeks/current")
```

ajax-gnatiuk-s commented on src/main/java/intellistart/interviewplanning/controllers/InterviewerController.kt:
```diff
@@ -0,0 +1,210 @@
+        authentication: Authentication
+    ): ResponseEntity<InterviewerSlotDtoResponse> {
+        val interviewerSlot = getInterviewerSlotsFromDto(interviewerSlotDtoRequest, authentication)
+        interviewerSlotDtoValidator
+            .validateUpdating(
+                interviewerSlot,
+                authentication, interviewerId, slotId
+            )
```
```suggestion
        interviewerSlotDtoValidator.validateUpdating(
            interviewerSlot,
            authentication, 
            interviewerId, 
            slotId
        )
```

ajax-gnatiuk-s commented on src/main/java/intellistart/interviewplanning/controllers/InterviewerController.kt:
```diff
@@ -0,0 +1,210 @@
+        authentication: Authentication,
+    ): InterviewerSlot {
+        val jwtUserDetails = authentication.principal as JwtUserDetails
+        val interviewerSlot = InterviewerSlot()
```
`.apply`?

ajax-gnatiuk-s commented on src/main/java/intellistart/interviewplanning/controllers/JwtAuthenticationController.kt:
```diff
@@ -0,0 +1,124 @@
+    private val jedis: JedisPooled
+) {
+    @Value("\${jwt.caching}")
+    private val jwtValidity: Long = 0
```
let's move to constructor?

ajax-gnatiuk-s commented on src/main/java/intellistart/interviewplanning/controllers/JwtAuthenticationController.kt:
```diff
@@ -0,0 +1,124 @@
+    @Value("\${jwt.caching}")
+    private val jwtValidity: Long = 0
+
+    private val logger: Logger = LogManager.getLogger(JwtAuthenticationController::class.java)
```
```suggestion
    companion object {
        private val logger: Logger = LogManager.getLogger(JwtAuthenticationController::class.java)
    }
```

Note: the companion object must be declared at the end of class

ajax-gnatiuk-s commented on src/main/java/intellistart/interviewplanning/controllers/JwtAuthenticationController.kt:
```diff
@@ -0,0 +1,124 @@
+        val userScopes: Map<FacebookScopes, String> = try {
+            facebookUtil.getScope(jwtRequest.facebookToken)
+        } catch (e: RestClientException) {
+            logger.warn("RestClientException was thrown", e)
```
The error message could be improved
```suggestion
            logger.warn("Failed to obtain user scope", e)
```

ajax-gnatiuk-s commented on src/main/java/intellistart/interviewplanning/controllers/JwtAuthenticationController.kt:
```diff
@@ -0,0 +1,124 @@
+            logger.warn("RestClientException was thrown", e)
+            throw SecurityException(SecurityExceptionProfile.BAD_FACEBOOK_TOKEN)
+        }
+        val email = userScopes[FacebookScopes.EMAIL]
+        val name = userScopes[FacebookScopes.NAME]
```
does it supposed that the map contains both keys for sure?

If yes then we can use `userScopes.getValue(FacebookScopes.NAME)`. It will throw if no such key, otherwise it will return non nullable value

ajax-gnatiuk-s commented on src/main/java/intellistart/interviewplanning/controllers/JwtAuthenticationController.kt:
```diff
@@ -0,0 +1,124 @@
+                UsernamePasswordAuthenticationToken(username, username)
+            )
+        } catch (e: BadCredentialsException) {
+            logger.warn("BadCredentialsException was thrown", e)
```
```suggestion
            logger.warn("Failed to authenticate user {}", username, e)
```

ajax-gnatiuk-s commented on src/main/java/intellistart/interviewplanning/controllers/JwtAuthenticationController.kt:
```diff
@@ -0,0 +1,124 @@
+        return ResponseEntity.ok(JwtResponse(jwt))
+    }
+
+    private fun authenticate(username: String?) {
```
```suggestion
    private fun authenticate(username: String) {
```

ajax-gnatiuk-s commented on src/main/java/intellistart/interviewplanning/controllers/JwtAuthenticationController.kt:
```diff
@@ -0,0 +1,124 @@
+            .loadUserByEmailAndName(email, name) as JwtUserDetails
+        val jwt = jwtUtil.generateToken(userDetails)
+        jedis.setex(jwtRequest.facebookToken, jwtValidity, jwt)
+        return ResponseEntity.ok(JwtResponse(jwt))
```
to complex logic for controller. A controller must be as dumb as possible. All this logic should moved to service layer

ajax-gnatiuk-s commented on src/main/java/intellistart/interviewplanning/controllers/JwtAuthenticationController.kt:
```diff
@@ -0,0 +1,124 @@
+    @GetMapping("/me")
+    fun getMyself(authentication: Authentication): ResponseEntity<*> {
+        val jwtUserDetails = authentication.principal as JwtUserDetails
+        val user = userService.getUserByEmail(jwtUserDetails.email)
+            ?: return ResponseEntity.ok(CandidateDto(jwtUserDetails.email))
+        return ResponseEntity.ok(user)
```
```suggestion
        return userService.getUserByEmail(jwtUserDetails.email)
            ?.let { user -> ResponseEntity.ok(user) }
            ?: ResponseEntity.ok(CandidateDto(jwtUserDetails.email))
```

ajax-gnatiuk-s commented on src/main/java/intellistart/interviewplanning/model/interviewerslot/InterviewerSlotDtoValidator.java:
```diff
@@ -65,37 +62,26 @@ public InterviewerSlotDtoValidator(PeriodService periodService,
    */
-  public InterviewerSlot validateAndCreate(InterviewerSlotDtoRequest interviewerSlotDtoRequest,
-                                           Authentication authentication, Long userId)
+  public void validateCreating(InterviewerSlot interviewerSlot,
+                                          Authentication authentication, Long userId)
```
```suggestion
  public void validateCreating(InterviewerSlot interviewerSlot,
                                               Authentication authentication, Long userId)
```

or 
```suggestion
  public void validateCreating(
          InterviewerSlot interviewerSlot,
          Authentication authentication, Long userId
  ) throws UserException, SlotException {
```

ajax-gnatiuk-s commented on src/main/java/intellistart/interviewplanning/model/interviewerslot/InterviewerSlotDtoValidator.java:
```diff
@@ -111,12 +97,10 @@ public InterviewerSlot validateAndCreate(InterviewerSlotDtoRequest interviewerSl
-  public InterviewerSlot validateAndUpdate(InterviewerSlotDtoRequest interviewerSlotDto,
-      Authentication authentication, Long userId, Long slotId)
+  public void validateUpdating(InterviewerSlot interviewerSlot,
+                                          Authentication authentication, Long userId, Long slotId)
```
strange formatting

ajax-huzei-a replied:
This class in will be rewritten to Kotlin, so I didn't focus much on formatting of all java classes that will be rewritten

ajax-gnatiuk-s reviewed (changes requested):
Good work. I like your neat code. Keep it going

ajax-huzei-a committed:
00001: KOTLIN MIGRATION
Move complex logic from JwtAuthenticationController to UserService and refactor formatting using "apply"

ajax-huzei-a committed:
00001: KOTLIN MIGRATION
Fix formatting for CheckStyle
